txpool/account.go

```diff
@@ -129,6 +132,73 @@ func (m *accountsMap) allTxs(includeEnqueued bool) (
 	return
 }
 
+func (m *accountsMap) reinsertProposed() {
+	m.Range(func(key, value interface{}) bool {
+		accountKey, ok := key.(types.Address)
+		if !ok {
+			return false
+		}
+
+		account := m.get(accountKey)
+
+		account.promoted.lock(true)
+		account.proposed.lock(true)
+		account.nonceToTx.lock()
+		account.nonceProposed.lock()
+
+		defer func() {
+			account.nonceProposed.unlock()
+			account.nonceToTx.unlock()
+			account.proposed.unlock()
+			account.promoted.unlock()
+		}()
+
+		if account.proposed.length() > 0 {
+			for {
+				tx := account.proposed.peek()
+				if tx == nil {
+					break
+				}
+
+				account.promoted.push(tx)
+				account.proposed.pop()
+			}
+			account.proposed.clear()
+
+			maps.Copy(account.nonceToTx.mapping, account.nonceProposed.mapping)
+			account.nonceProposed.reset()
+		}
+
+		return true
+	})
+}
+
+func (m *accountsMap) clearProposed() {
+	m.Range(func(key, value interface{}) bool {
+		accountKey, ok := key.(types.Address)
+		if !ok {
+			return false
+		}
+
+		account := m.get(accountKey)
+
+		account.proposed.lock(true)
+		account.nonceProposed.lock()
+
+		defer func() {
+			account.nonceProposed.unlock()
+			account.proposed.unlock()
+		}()
+
+		if account.proposed.length() > 0 {
```

Comment:
I don't think we need this check. Just call clear and reset, they are not expensive funcitons.

Reply:
Maybe it's better to keep it. I had same dilemma, what is more expensive, len() or clear() and reset(). Reset has memory allocation for slice, so in the situation when we have thousands of accounts in the pool doing slice allocation for every account could be slower then checking queue length.